bug-fix/fix-ui-padding

ahmedNaser7 opened the pull request:
feat: round rating before calculating average
fix padding in request details 
hide voice icon in message details
hide app bar"s details icon in request details

`roundFloat` to round float values to one decimal place. This function is then used in `UserRemoteDataSourceImpl` to round individual ratings before calculating the average rating for a craftsman.

ahmedNaser7 committed:
feat: round rating before calculating average

This commit introduces a utility function `roundFloat` to round float values to one decimal place. This function is then used in `UserRemoteDataSourceImpl` to round individual ratings before calculating the average rating for a craftsman.

ahmedNaser7 committed:
fix : rating value & add padding for image

- Fix: The rating value now correctly displays floating-point numbers.
- UI: Added padding to the bottom of images for better visual spacing.

ahmedNaser7 committed:
delete app bar from request details

ahmedNaser7 committed:
remove the voice feature for now

ahmedNaser7 committed:
remove comment for voice feature in message text field

ahmedNaser7 committed:
fix: remove unused imports

ahmedNaser7 committed:
Merge branch 'develop' into bug-fix/fix-ui-padding

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/data/source/remote/user/UserRemoteDataSourceImpl.kt:
```diff
@@ -215,20 +216,23 @@ class UserRemoteDataSourceImpl(
-                    ratings.mapNotNull { it }.average().toFloat()
+                    ratings.mapNotNull { it?.roundFloat() }
+                        .average()
+                        .toFloat()
```
roundFloat

ahmedNaser7 replied:
fix the floating numbers , like 2,555555 to 2,5 only

Joseph-Sameh-0 replied:
make the round at the end not on every value

ahmedNaser7 replied:
no problem

ahmedNaser7 committed:
Merge branch 'develop' into bug-fix/fix-ui-padding

# Conflicts:
#	app/src/main/java/com/paris_2/san3a/presentation/screen/requestDetails/craftsman/CraftsManRequestDetailsScreen.kt

ahmedNaser7 committed:
resolve conflicts

ahmedNaser7 committed:
hide the button action of requests of craftsman when is accepted

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/requestDetails/craftsman/CraftsManRequestDetailsScreen.kt:
```diff
@@ -200,7 +200,7 @@ fun CraftsmanRequestDetailsContent(
                         )
                         CraftsManOffer(
                             addShadow = true,
-                            showActionButtons = state.yourOffer == state.acceptedOffer,
```
it's only if accepted
there is another condition if completed or canceled

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/requestDetails/craftsman/CraftsManRequestDetailsScreen.kt:
```diff
@@ -200,7 +200,7 @@ fun CraftsmanRequestDetailsContent(
                         CraftsManOffer(
                             addShadow = true,
-                            showActionButtons = state.yourOffer == state.acceptedOffer,
+                            showActionButtons = state.yourOffer != state.acceptedOffer,
```
this part is wrong 
 the state.yourOffer == state.acceptedOffer is right 
but && the status is on going

ahmedNaser7 committed:
fix: calculation of average ratings and UI adjustments for craftsman offers

This commit addresses two main areas:

1.  **Average Rating Calculation:**
    *   Corrected the calculation of average ratings in `UserRemoteDataSourceImpl.kt` by ensuring `roundFloat()` is applied after averaging, not before.

2.  **Craftsman Offer UI:**
    *   Modified the visibility condition for action buttons in `CraftsManRequestDetailsScreen.kt` to only show them if the craftsman's offer is the accepted one and the request status is "ONGOING".
    *   Temporarily hid the rate in `CraftsmanOffer.kt` as it's not yet fully implemented.
    *   Updated the `craftsman_reviews` string resource in both English and Arabic to remove the review count, simplifying the display in `CraftsmanOfferDetails.kt`.
